### PR TITLE
Added update of URL while scrolling

### PR DESCRIPTION
**Pull request by morkro**
This will update the URL to the current visible section (e.g. `sass-guidelin.es/#about-sass`) while scrolling.

I turned some of the functionality into a `helper.js`. Right now the scrolling code from `window.Sidebar` and `window.UrlManager` are identical. Might make sense to turn that into another helper function because DRY.

I've used `window.location.hash` and not `window.history.pushState()` is I felt it would feel weird to the user if the "go back"-button would jump between sections the site. What do you think?

### CODE REVIEW

File: _includes/critical.css [hunk 1]
```diff
@@ -1 +1 @@
...
...0;background-color:transparent;-webkit-transform:scale(1.25)}.aside__radio-list{list-style:none;padding:0;margin:-.5rem;overflow:hidden}.aside__radio-list label{display:inline-block;cursor:pointer}.aside__radio-list input{-webkit-transform:translate(-1px,-1px)}.aside__radio-list li{margin:0;float:left;width:50%;padding:0 .5rem}@media (min-width:975px){.baseline{max-width:975px}.toc{overflow-x:hidden;overflow-y:auto;position:absolute;top:0;left:0;bottom:0;width:300px;z-index:10;font-size:.8em;color:#444;background-color:#fff}.toc ul{margin:0;padding:12px 10px;list-style:none}.toc li{border-bottom-width:1px;border-bottom-style:solid;border-bottom-color:#eee;padding:8px 0;margin-bottom:0}.toc a{text-decoration:none}.toc h1,.toc li:nth-of-type(3){display:none}}@media print{*{color:#000!important;text-shadow:none!important;box-shadow:none!important}a[href^=http]::after{content:' (',attr(href),')';text-decoration:none;font-style:italic;opacity:.7;-webkit-hyphens:auto;overflow-wrap:break-word;word-wrap:break-word}.anchor-link,.aside,.button-ui--back,.toc,.translation-warning,[data-syntax=sass],svg{display:none}.header{border-bottom-width:0}h1{margin-top:3rem}h1,h2{page-break-after:avoid}}
```
Comment (HugoGiraudel) [inline _includes/critical.css]: Please discard this.

File: assets/js/main.js [hunk 2]
```diff
@@ -1,14 +1,21 @@
 document.addEventListener('DOMContentLoaded', function (event) {
+  var headings = document.querySelectorAll('.chapter:not(.toc) > h1[id]');
+
   var app = new window.App({
     languagePicker: document.getElementById('language-picker')
   });
 
   var sidebar = new window.Sidebar({
     addOffsetView: 50,
```
Comment (HugoGiraudel) [inline assets/js/main.js]: This should probably be stored in a variable as well.

File: assets/js/helper.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+(function (global) {
+	'use strict';
+
+	global.getDocumentHeight = function () {
```
Comment (HugoGiraudel) [inline assets/js/helper.js]: Do you feel like setting up a little bundler like Browserify (or something even lighter would be better)? I’m not too concerned about JS size, and we are starting to do more and more in JS so would be nice to be able to write clean scoped JS.

File: assets/js/UrlManager.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      this.updateOffset = Array.prototype.slice.call(this.updateOn)
+        .map(function (item) {
+          return [ item, global.getOffset(item) ];
+        });
+
+      this.initialize();
+   };
+
+   UrlManager.prototype.initialize = function () {
+      console.log(this);
```
Comment (HugoGiraudel) [inline assets/js/UrlManager.js]: Come on, Moritz… :x

File: assets/js/helper.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+(function (global) {
+	'use strict';
```
Comment (HugoGiraudel) [inline assets/js/helper.js]: Wait too much indented. :D

File: assets/js/UrlManager.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      this.isLargerThanMobile && this.evalUserPosition();
+
+      global.addEventListener('scroll', this.evalUserPosition.bind(this), false);
+      global.addEventListener('resize', evalClientResolution, false);
+   };
+
+   UrlManager.prototype.evalUserPosition = function () {
```
Comment (HugoGiraudel) [inline assets/js/UrlManager.js]: This whole thing is also duplicated in Sidebar.js.

Comment (morkro) [inline assets/js/UrlManager.js]: Yes I know. I've told you :D Will update this, after the bundler is available.

File: assets/js/src/UrlManager.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
+(function (global) {
+   'use strict';
+
+   function evalClientResolution () {
```
Comment (HugoGiraudel) [inline assets/js/src/UrlManager.js]: This is already in the helpers.

File: assets/js/src/helpers.js [hunk 8]
```diff
@@ -1,21 +1,17 @@
 export default {
-  evalClientResolution: function evalClientResolution (size) {
-    return matchMedia('(min-width: ' + size + 'px)').matches;
+  evalClientResolution (size) {
```
Comment (HugoGiraudel) [inline assets/js/src/helpers.js]: I didn’t know rollup compiled ES6.

**Comment by morkro** [comment]
Too many changes done in a different PR. Will close this one, and open a cleaner one later.

**Pull request closed without merging**